File: src/leetcode/trees/preorder/SumRoottoLeafNumbers.java
package leetcode.trees.preorder;

import leetcode.trees.TreeNode;

public class SumRoottoLeafNumbers {
    public int sumNumbers(TreeNode root) {
        return cal(root, 0);
    }

    private int cal(TreeNode root, int sum) {
        if (root == null) {
            return 0;
        }
        int currentSum = sum * 10 + root.val;
        if (root.left == null && root.right == null) {
            return currentSum;
        }
        return cal(root.left, currentSum) + cal(root.right, currentSum);

    }

}

File: src/leetcode/trees/preorder/SumRoottoLeafNumbers.js
var sumNumbers = function(root) {
    return cal(root, 0);
};

const cal = (root, sum) => {
    if (!root) {
        return 0;
    }
    let currentSum = sum*10 + root.val;
    if (!root.left && !root.right) {
        return currentSum;
    }
    return cal(root.left, currentSum) + cal(root.right, currentSum);
}